zarr/core.py

```diff
@@ -1835,6 +1837,48 @@ def bytestr(n):
 
         return items
 
+    def hexdigest(self, hashname="sha1"):
+        """
+        Compute a checksum for the data. Default uses sha1 for speed.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> z = zarr.empty(shape=(10000, 10000), chunks=(1000, 1000))
+        >>> z.hexdigest()
+        '096814656b4d26c9301537b76fc3a72d8cb9c71b'
+        >>> z = zarr.zeros(shape=(10000, 10000), chunks=(1000, 1000))
+        >>> z.hexdigest()
+        '77cac405bc1c9b3bcddae92c49f63102b41cdb4f'
+        >>> z = zarr.zeros(shape=(10000, 10000), dtype="u1", chunks=(1000, 1000))
+        >>> z.hexdigest()
+        'ccc9cea31d1cbb8596fc0c076d437223ddb024a2'
+        """
+
+        h = hashlib.new(hashname)
+
+        for i in itertools.product(*[range(s) for s in self.cdata_shape]):
+            try:
+                d = self.chunk_store[self._chunk_key(i)]
+                h.update(d)
+            except KeyError:
+                pass
+
+        filters = self.filters if self.filters else []
+        for meta in [self.fill_value, self.dtype, self.shape, self.order] + filters:
+            r = repr(meta)
+            try:
+                h.update(r)
+            except TypeError:
+                h.update(r.encode())
```

Comment:
Instead of this you could just pull out '.zarray' from the store and hash that.

Reply:
Sounds like a much better idea. Can we rely on `.zarray` always being there or should we add some logic for when it is missing?

Reply:
It will always be there, an array isn't an array if it doesn't have a .zarray.